charmhelpers/contrib/openstack/amulet/deployment.py

```diff
@@ -303,8 +303,15 @@ def get_ceph_expected_pools(self, radosgw=False):
         test scenario, based on OpenStack release and whether ceph radosgw
         is flagged as present or not."""
 
-        if self._get_openstack_release() >= self.trusty_kilo:
-            # Kilo or later
+        if self._get_openstack_release() >= self.xenial_pike:
+            # Pike and later
+            pools = [
+                'cinder-ceph',
+                'glance'
+            ]
+        elif (self._get_openstack_release() >= self.trusty_kilo and
+              self._get_openstack_release() <= self.zesty_ocata):
+            # Kilo through Ocata
```

Comment:
I had to stare at this for a while to understand it.

The 2nd elif can be written a little more nicely as:

```python
    elif self.trust_kilo <= self._get_openstack_release() <= self.zesty_ocata:
        ...
```
But now the code looks a little confusing.  I'd like to suggest switching the order of the `if` statements so that they go in order of OpenStack releases.  e.g. start with the juno or earlier, then pick off the inbetween ones and finally have the latest so that it reads chronologically.

Reply:
Confirmed, the test helper code itself generally doesn't have unit test coverage.

Reply:
Simplified and re-ordered per recommendations - thank you.  See updated PR.